src/ContractInteractor.ts

```diff
@@ -919,20 +945,25 @@ export default class ContractInteractor {
 
     async getERC20Token(
         address: string,
-        options?: TokenOptions
-    ): Promise<Token> {
-        const token = await this._createERC20(address);
+        options?: Partial<ERC20Options>
+    ): Promise<ERC20Token> {
+        const instance = await this._createERC20(address);
+        if (!options || Object.values(options).every((x) => x === false)) {
```

Comment:
Why is it superfluous? Why do we need to instantiate and wait for `promise.all` if not necessary?

Reply:
not sure I get you correctly, but if I do than I think you might be misunderstanding me.
What I tried to say there is that we don't need to [retrieve properties, iterate, compare] on each option, because we can just check them directly for each statement. This should result in less calls overall as it would only access the instance if a given option is set to true.
Please tell me if I'm not getting your point or viceversa

Reply:
I do agree with you, we don't really need the `if (!options || Object.values(options).every((x) => x === false)) {` because we check for each property later on. 👍🏽

Reply:
updated :)